includes/ziti/model_collections.h

```diff
@@ -80,8 +78,37 @@ for (var(e,line) = model_map_iterator(map), k = model_map_it_key(var(e,line)), v
 
 #define MODEL_MAP_FOR(it, m) for(model_map_iter it = model_map_iterator(&(m)); (it) != NULL; (it) = model_map_it_next(it))
 
+typedef struct model_list_s {
+    struct model_list_impl_s *impl;
+} model_list;
+
+typedef void *model_list_iter;
+
+ZITI_FUNC size_t model_list_size(const model_list *l);
+
+ZITI_FUNC void model_list_append(model_list *l, void *el);
+
+ZITI_FUNC void *model_list_pop(model_list *l);
```

Comment:
looking at the impl - should this be model_list_pop_element? it also calls model_list_it_remove which ... without doc i didn't expect ? :)

Reply:
Pop operation is a standard name for removing top of the list/stack/etc.